### PR TITLE
Fix `trackingFree()` - do not add a not removed entry

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Fix `trackingFree()` - do not add a not removed entry.
Do not add memory back to the tracker,
if it had not been removed.

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Review by bratpiorka — APPROVED** [review]

File: src/provider/provider_tracking.c [hunk 1]
```diff
@@ -366,6 +372,12 @@ static umf_result_t trackingFree(void *hProvider, void *ptr, size_t size) {
     ret = umfMemoryProviderFree(p->hUpstream, ptr, size);
     if (ret != UMF_RESULT_SUCCESS) {
         LOG_ERR("upstream provider is failed to free the memory");
+        // Do not add memory back to the tracker,
+        // if it had not been removed.
+        if (ret_remove != UMF_RESULT_SUCCESS) {
```
Comment (lukaszstolarczuk) [inline src/provider/provider_tracking.c:377]: `Using uninitialized memory 'ret_remove'`

**Review by vinser52 — APPROVED** [review]

File: src/provider/provider_tracking.c [hunk 2]
```diff
@@ -332,6 +337,7 @@ static umf_result_t trackingAllocationMerge(void *hProvider, void *lowPtr,
 
 static umf_result_t trackingFree(void *hProvider, void *ptr, size_t size) {
     umf_result_t ret;
+    umf_result_t ret_remove;
```
Comment (bratpiorka) [inline src/provider/provider_tracking.c:340]: as Luke wrote, please initialize

File: src/provider/provider_tracking.c (hunk 1 above)
Comment (ldorau) [reply src/provider/provider_tracking.c:377]: Done

File: src/provider/provider_tracking.c (hunk 2 above)
Comment (ldorau) [reply src/provider/provider_tracking.c]: Done

**Review by lukaszstolarczuk — COMMENTED** [review]
👍

**Pull request merged**